### ISSUE TITLE
Initial Confidence Metric

### ISSUE THREAD

**Issue by nluedtke** [nluedtke/LineIt#10]
Variance Calculation for Offense, Defense, Differential.

BLOCKed by #2, #4

**Comment by nluedtke** [comment]
No longer blocked

**Comment by nluedtke** [comment]
Do you want to spit out the variance of Offense Rating, Defensive Rating, and IPD across the whole league?

**Comment by tnees** [comment]
Each team should have its own variance ratings for each of the three categories. A league wide variance would be nice, but not necessary.

**Comment by nluedtke** [comment]
Ok before i start coding to clarify:

So if an opponent allows an average of 22 points, and Team A scored 40, they get a +18 for the game. If the next opponent allows an average of 27, and Team A gets 12, they get a -15.

In that example, you are thinking that there is a separate set of numbers. [18, -15] for offense that you want a variance based off of?

**Comment by tnees** [comment]
It's easiest to look at a longer than two game example for me to parse it.

So, Team A plays a season of 6 games. The algorithm runs, and establishes the defensive rating for each of their six opponents (DR1, DR2...DR6). So, we establish the single game values to average out for Team A's offensive rating, using (ActualScored1-DR1, AS2-DR2...AS6-DR6) creating an array of (RatedGame1, RG2...RG6). We then want to find the variance, so we're going to take (|RG1-OffensiveRating|  + |RG2-OR| + ... + |RG6-OR|)/6. This gives us the average variance per game. Once we have that, we can use it to generate a lot of other metrics to examine, such as standard deviations, and helps us establish a Gauss distribution if we can.

Basically, this is the key step that's going to inform additional metrics; based on the numbers here I'll have a much better idea exactly which math functions can be applied easily to the higher level outputs to narrow our eventual line as much as possible.

**Comment by nluedtke** [comment]
The distributions are easy to do, just trying to get at the data you want me to throw into it. Which i think i have now.

**Comment by nluedtke** [comment]
Initial Rating = League_Avg - Team_Avg
AdjRating (for a given week on first pass) = Actual_Scored - OpposingTeam_Rating

I got the variance part now, but do you want to use OpposingTeam's rating or OpposingTeam's avg allowed/scored?
